### ISSUE TITLE
Inconsistent naming patterns of champions in ddragon url's

### ISSUE THREAD

**Issue by singerm0** [RiotGames/developer-relations#7]
Due to the complexities of champion names not all champ names can be directly put into a url. However Individual names seem random at times. The general scheme seems to be:

- If a name has no special characters do not change anything. ex: Aatrox, Annie, Nautilus
- If there is a space or period, delete the space/period and do not change the case ex: AurelionSol, DrMundo, MasterYi.
- If there is a ' in the name randomly do one of two things, delete it and make the second part lower case or delete it and keep the second part as upper case ex: **KogMaw, Khazix, Chogath, RekSai, or Velkoz.**

Also, there are a few exceptions to the general scheme in Wukong being MonkeyKing, LeBlanc being leblanc, and Nunu & Willump being Nunu.

**Comment by molenzwiebel** [comment]
This is because DDragon uses the champion key instead of it's name. You can simply use the key, which should be in `champion.json`. While I agree that it would be nice to have this consistent, the devrel team is not in charge of DDragon and would only be able to relay the sentiment to the champions team responsible for all static data. Note that such a change would also nuke any chance of backwards compat.

**Comment by riottuxedo** [comment]
You should use the `champion.json` file to determine the id of the champion and then use the id to look up a champions data. This is best illustrated with Wukong.

http://ddragon.leagueoflegends.com/cdn/8.24.1/data/en_US/champion.json
```
"MonkeyKing": {
  "id": "MonkeyKing",
  "key": "62",
  "name": "Wukong",
  "title": "the Monkey King",
  "image": {
    "full": "MonkeyKing.png",
    ... // additional image details
  }
}
```

Above you can see that Wukong's key is `MonkeyKing` which you can then use to look up Wukong's champion data.
http://ddragon.leagueoflegends.com/cdn/8.24.1/data/en_US/champion/MonkeyKing.json

You'll also notice that the name of the full image file is `MonkeyKing.png` which you can find here:
http://ddragon.leagueoflegends.com/cdn/6.24.1/img/champion/MonkeyKing.png

**Comment by KonssnoK** [comment]
Following @querijn notes on 
https://github.com/RiotGames/developer-relations/issues/83
how am i supposed to resolve `FiddleSticks` from match data to 
```
"Fiddlesticks":{"version":"13.24.1","id":"Fiddlesticks","key":"9","name":"Fiddlesticks","title":"the Ancient Fear","blurb":"Somethin...","info":{"attack":2,"defense":3,"magic":9,"difficulty":9},"image":{"full":"Fiddlesticks.png","sprite":"champion1.png","group":"champion","x":48,"y":0,"w":48,"h":48},"tags":["Mage","Support"],"partype":"Mana","stats":{"hp":650,"hpperlevel":106,"mp":500,"mpperlevel":28,"movespeed":335,"armor":34,"armorperlevel":4.7,"spellblock":30,"spellblockperlevel":1.3,"attackrange":480,"hpregen":5.5,"hpregenperlevel":0.6,"mpregen":8,"mpregenperlevel":0.8,"crit":0,"critperlevel":0,"attackdamage":55,"attackdamageperlevel":2.65,"attackspeedperlevel":2.11,"attackspeed":0.625}},
```
?

Naming is mismatched to champion.json too.. :\

**Comment by Derek4aty1** [comment]
It seems there is some misunderstanding across multiple GitHub issues on this Fiddlesticks topic. I'll try and fully explain the issue since I discovered it in my own time as well and would like to see it fixed (TL;DR at the end).

Take a look Fiddlesticks champion.json below. The champion id (located at data.Fiddlesticks.id) is "Fiddlesticks" with a lower case 's':
https://ddragon.leagueoflegends.com/cdn/14.14.1/data/en_US/champion/Fiddlesticks.json

We are supposed to use the champion id to retrieve champion assets, as others have mentioned.
**The problem with Fiddlesticks is that all of the splash, loading, and ability (excluding passive) art use "FiddleSticks" with an uppercase 's' instead...this doesn't match the champion id.**
Refreshed splash art for Fiddlesticks:
https://ddragon.leagueoflegends.com/cdn/img/champion/splash/FiddleSticks_0.jpg

If you use the champion id with the lowercase 's' as intended from the champion.json for either splash or loading art, then the outdated art is returned (Fiddlesticks has been in the game since the very beginning and a few years ago his art got refreshed):
https://ddragon.leagueoflegends.com/cdn/img/champion/splash/Fiddlesticks_0.jpg

For the more recent skins, there is no outdated art to pull so if you use the champion id with the lowercase 's' as intended from the champion.json you get an "Access Denied" error (I assume the file doesn't exist):
https://ddragon.leagueoflegends.com/cdn/img/champion/splash/Fiddlesticks_37.jpg

For the passive and abilities, there are effect id strings (located at data.Fiddlesticks.passive.image.full and data.Fiddlesticks.spells[index].image.full respectively)  within the champion.json that can be used to retrieve the image so these are less of a problem, but they should still match the naming convention with the lowercase 's' in my opinion...
The passive matches already so that is good at least!
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/passive/FiddlesticksP.png
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/spell/FiddleSticksQ.png
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/spell/FiddleSticksW.png
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/spell/FiddleSticksE.png
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/spell/FiddleSticksR.png

The square icon matches the champion id as intended so that is also good:
https://ddragon.leagueoflegends.com/cdn/14.14.1/img/champion/Fiddlesticks.png

TL;DR 
Fiddlesticks champion id is "Fiddlesticks" but most assets use "FiddleSticks" meaning you cannot dynamically fetch Fiddlesticks' assets using the id as intended.
